### DISCUSSION TITLE
Test Breaks After Deep Link Re-direct back to App (Auth Flow Problem)

### DISCUSSION THREAD

**Discussion by sonnyvesali** [leancodepl/patrol#1307] [Q&A]
Hi, thank you for the wonderful documentation and library, but I'm having an issue testing my authentication flow in my app. The first problem I have is the fact that in my authentication flow, the user on android at least is re-directed to their default browser (usually chrome), they do the google sign in, and THEN they are supposed to be re-directed back to the app, and the app should automatically navigate to the home screen. The Problem is the fact that when you manually go through the auth flow, you do the sign in, and then you get re-directed back to the app, but what happens is that, it's almost like a flash re-direct happens and then it just stays on the google sign in page and just keeps trying to re-direct, I think it may have to do with the app getting deleted after that and the deep link re-direct has nowhere to go.

```dart
// patrol test code
// removed imports for brevity

void main() {
  patrolTest(
    'Should login user with Google on iOS & Android',
    nativeAutomation: true,
    framePolicy: LiveTestWidgetsFlutterBindingFramePolicy.benchmarkLive,
    ($) async {
     // removed app setup for brevity
      final googleEmail = dotenv.env["TEST_GOOGLE_EMAIL"] ?? '';
      final googlePassword = dotenv.env["TEST_GOOGLE_PASSWORD"] ?? '';
      await $(K.signInWithGoogleButton).tap();
      late String appId;
      if (Platform.isIOS) {
        appId = 'com.apple.mobilesafari';
        await $.native.enterTextByIndex(googleEmail, index: 1, appId: appId);
        await $.native.tap(Selector(textContains: "Next"), appId: appId);
        await $.native.enterTextByIndex(googlePassword, index: 1, appId: appId);
        await $.native.tap(Selector(textContains: "Next"), appId: appId);
      } else {
        appId = 'com.android.chrome';
        await $.native.enterText(Selector(focused: true), text: googleEmail, appId: appId);
        await $.native.tap(Selector(textContains: "Next"), appId: appId);
        await $.native.enterText(Selector(focused: true), text: googlePassword, appId: appId);
        await $.native.tap(Selector(textContains: "Next"), appId: appId);
        // await $.native.tap(Selector(text: "Sign Out"));  
       // ^^  this makes it so that the app doesn't get deleted immediately, but it throws a `Not Found` exception
      }
      expect(find.text("Sign Out"), findsOneWidget); // <=== Not working even when it would re-direct to the app
      // await $.native.pressHome();
    },
  );
}
```
Video Evidence (assume the in app google button got pressed, video started late): 

https://github.com/leancodepl/patrol/assets/79886185/ba94f391-6d77-4c5f-a6d2-3f90e2d0fd99

A side question that is less urgent is that what's wrong with my iOS testing logic, since the field isn't activated by default on safari, it needs to be activated. I'm pretty sure 0 enters it into the link bar, and 1 is the next test field so what's wrong with that. Sorry to pack this with so much, but I would love to submit a PR to the patrol docs for a cookbook section of writing tests like this, with my learnings as a way of giving back, thanks again.

**Comment by bartekpacia** [comment]
Hi @sonnyvesali, thanks for your kind words and willingness to contribute. We have a few docs-related issues open, e.g. #287 and #859.

Unfortunately, I don't know what could be causing your problem with Google Sign-in _flashing_ for a short instant and then being stuck.

Regarding the second question – I also don't know :( Unfortunately our webview feature is not stable (see #1139), but we'll get back to it soon and make it much better.

 It'd be best if you could create an issue [minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example).
